src/unix/signal.c

```diff
@@ -167,6 +172,10 @@ static void uv__signal_handler(int signum) {
 
     if (r != -1)
       handle->caught_signals++;
+
+    if ((handle->flags & UV_SIGNAL_ONE_SHOT) &&
+        (uv__signal_first_handle(handle->signum) == NULL))
+      uv__signal_unregister_handler(signum);
```

Comment:
>We could map that to .on and .once in node, with .on subsuming .once.

While looking into this... is there a way to know in `node` when a `once` listener is attached? By looking into the `newListener` event documentation I couldn't find it.